bee-node/src/tools/benchmark_cpu.rs

```diff
@@ -0,0 +1,135 @@
+use std::{
+    sync::{
+        atomic::{AtomicU64, Ordering},
+        Arc,
+    },
+    thread::{self},
+    time::{self, Duration},
+};
+
+use bee_crypto::ternary::sponge::{CurlP, CurlPRounds, Sponge, BATCH_SIZE};
+use bee_pow::providers::miner::MinerCancel;
+use bee_ternary::{
+    b1t6::{self},
+    T1B1Buf, TritBuf,
+};
+use structopt::StructOpt;
+use thiserror::Error;
+
+#[derive(Debug, Error)]
+pub enum BenchmarkCPUError {
+    #[error("{0}")]
+    Io(#[from] std::io::Error),
+}
+
+#[derive(Clone, Debug, StructOpt)]
+pub struct BenchmarkCPUTool {
+    threads: Option<usize>,
+}
+
+pub fn exec(tool: &BenchmarkCPUTool) {
+    let threads = match tool.threads {
+        Some(threads) => threads,
+        None => num_cpus::get(),
+    };
+    let duration = Duration::from_secs(60);
+
+    println!("Benchmarking CPU with {} threads", threads);
+
+    let cancel = MinerCancel::new();
+    let cancel_2 = cancel.clone();
+    let cancel_3 = cancel.clone();
+    let counter = Arc::new(AtomicU64::new(0));
+    let counter_2 = counter.clone();
+
+    let time_start = std::time::Instant::now();
+
+    let pow_digest: [u8; 32] = rand::random();
+
+    let mut workers = Vec::with_capacity(threads + 2);
+
+    //Stop if the timeout has exceeded
+    let time_thread = thread::spawn(move || {
```

Comment:
maybe these should be `tokio::task::spawn` instead

Reply:
Is there a difference in performance? It was the only thing I knew back then. As I said this code is quite old 😆

Reply:
Well... maybe. Tokio's task are closer to goroutines than OS threads. I guess it depends on how much context switching is the OS doing with the threads or something.